### PR TITLE
Doc Anchor Fixes

### PR DESCRIPTION
**Pull request by dblotsky**
- Moving anchor creation out of ToC logic. 
- Moving ID creation into JS because RedCarpet doesn't support Unicode headings. 
- Making permalink icons a bit smaller and making them always show on small and medium screens.

### CODE REVIEW

**Comment by dblotsky** [comment]
@nikhilkh @riknoll @sarangan12 Please review when you folks get a chance.

File: www/static/js/docs.js [hunk 1]
```diff
@@ -17,49 +17,81 @@
...
 
+    function anchorFor(id) {
+        var anchor       = document.createElement("a");
+        anchor.className = "fragment-anchor";
+        anchor.id        = id;
+        return anchor;
+    }
+
+    // go through all headings in the content and add some links
+    $('#page-toc-source h1, #page-toc-source h2').each(function (i, heading) {
```
Comment (nikhilkh) [inline www/static/js/docs.js]: Does this still couple it to the TOC - `#page-toc-source`? Also, we should add links for all headings h3+ should be included.

File: www/static/js/docs.js [hunk 2]
```diff
@@ -17,49 +17,81 @@
...
+            $(heading).append(permalinkTo(headingId));
+        }
+    });
+
     // Table of Contents
     $('#page-toc').toc({
         'selectors':         'h1,h2', // elements to use as headings
         'container':         '#page-toc-source', // element to find all selectors in
-        'prefix':            'link-', // prefix for anchor tags and class names
+        'prefix':            'toc-', // prefix for anchor tags and class names
```
Comment (nikhilkh) [inline www/static/js/docs.js]: can we remove the prefix?

File: www/static/js/docs.js (hunk 1 above)
Comment (riknoll) [inline www/static/js/docs.js]: +1 The Android guide definitely requires referencing H3+

**Comment by nikhilkh** [comment]
I'm assuming you grep'd for any `#link-` anchor links on our site that need to be changed now as we change the slugify scheme and prefix.

File: www/static/js/docs.js (hunk 2 above)
Comment (dblotsky) [inline www/static/js/docs.js]: Yep, I removed it from the code. I'll make it empty to make that more clear.

File: www/static/js/docs.js (hunk 1 above)
Comment (dblotsky) [inline www/static/js/docs.js]: Added H3.

**Comment by dblotsky** [comment]
@nikhilkh: good catch! Fixed links with `#link-` in them.

**Comment by dblotsky** [comment]
@nikhilkh I kept `#page-toc-source` because it's just an ID, which just happens to be used by the ToC too. If the ID ever changes or is moved, all uses of it should be examined and then we can create another one if required. The motivation to not touch it unless necessary is the fact that changing it will lead to a massive SVN commit (and right now it's not strictly necessary).

**Comment by nikhilkh** [comment]
LGTM. Thanks, Dmitry for fixing this! I just tested it and it works very well!

**Pull request merged**